src/System.Linq/src/System/Linq/Set.cs

```diff
@@ -191,6 +191,27 @@ internal List<TElement> ToList()
         }
 
         /// <summary>
+        /// Creates a hashset from the items in this set.
+        /// </summary>
+        /// <returns>A list of the items in this set.</returns>
+        internal HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer)
+        {
+#if DEBUG
+            Debug.Assert(!_haveRemoved, "Optimised ToHashSet cannot be called if Remove has been called.");
+#endif
+            int count = _count;
+
+            HashSet<TElement> hashSet = new HashSet<TElement>(count, comparer);
```

Comment:
This is also one of the few places where you may want to keep the count. I don't know how common it would be for people to do `Union` with one comparer and `ToHashSet` on another (I think most of the time, people would stick to the default comparer). /cc @JonHanna